### PR TITLE
Updated projen and added homepage to blueprint builder

### PR DESCRIPTION
**Pull request by pranavdo**
### Issue

Issue number, if available, prefixed with "#"

### Description

What does this implement? Explain your changes.
It adds homepage field to blueprint builder who default for now is set to aws.amazon.com. This homepage will be used in the future to retrieve homepage url from the blueprint settings page and other locations to be redirected to source repo for a blueprint.

### Testing

How was this change tested?
Through npm run build and other yarn build instructions listed in the Readme.

### Additional context

Add any other context about the PR here.
results of `npm run build` - 
```
 YN0000: [@amazon-codecatalyst/blueprints]: Process started
➤ YN0000: [@amazon-codecatalyst/blueprints]: 👾 build » default | ts-node --project tsconfig.dev.json .projenrc.ts
➤ YN0000: [@amazon-codecatalyst/blueprints]: 👾 build » compile | tsc --build
➤ YN0000: [@amazon-codecatalyst/blueprints]: 👾 build » test » eslint | eslint --ext .ts,.tsx --fix --no-error-on-unmatched-pattern src test build-tools projenrc .projenrc.ts
➤ YN0000: [@amazon-codecatalyst/blueprints]: 👾 build » package | mkdir -p dist/js
➤ YN0000: [@amazon-codecatalyst/blueprints]: 👾 build » package | mv $(npm pack) dist/js/
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice 
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice 📦  @amazon-codecatalyst/blueprints@0.3.147
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice === Tarball Contents === 
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice 546B   .gitattributes
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice 11.4kB LICENSE       
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice 2.7kB  lib/index.js  
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice 2.3kB  package.json  
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice 275B   README.md     
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice 1.1kB  .projenrc.ts  
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice 600B   lib/index.d.ts
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice === Tarball Details === 
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice name:          @amazon-codecatalyst/blueprints           
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice version:       0.3.147                                   
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice filename:      amazon-codecatalyst-blueprints-0.3.147.tgz
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice package size:  6.4 kB                                    
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice unpacked size: 18.8 kB                                   
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice shasum:        4309a222be8d72fdb5b99d28da44322347680a17  
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice integrity:     sha512-beSIWofsJNM/v[...]apkxhVBoZYmOQ==  
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice total files:   7                                         
➤ YN0000: [@amazon-codecatalyst/blueprints]: npm notice 
➤ YN0000: [@amazon-codecatalyst/blueprints]: Process exited (exit code 0), completed in 4s 65ms
➤ YN0000: Done in 1m 29s
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/blueprints/sam-serverless-app/.gitattributes
Comment (aggagen) [inline packages/blueprints/sam-serverless-app/.gitattributes:1]: I'm not sure we're ready to remove `sam-serverless-app` from this repo. Can we remove that from this PR?

File: packages/blueprints/blueprint-builder/.projenrc.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ const blueprint = new ProjenBlueprint({
   description: 'This blueprint creates a project which you can use to build and publish a custom blueprint to your space.',
   /* The "name" in package.json. In form @amazon-codecatalyst/:organization.:name */
   packageName: '@amazon-codecatalyst/blueprints.blueprint-builder',
-  publishingOrganization: 'blueprints',
```
Comment (aggagen) [inline packages/blueprints/blueprint-builder/.projenrc.ts:29]: this should stay as `blueprints`

File: packages/blueprints/blueprint-builder/package.json [hunk 2]
```diff
@@ -24,8 +24,8 @@
     "build:lib": "rm -rf ./lib/ && yarn build && yarn blueprint:build-ast && yarn blueprint:validate-options",
     "blueprint:package": "yarn build:lib && yarn blueprint:synth --cache --clean-up false && yarn package",
     "npm:publish": "npm publish dist/js/*.tgz",
-    "blueprint:preview": "yarn bump:preview && yarn blueprint:package && blueprint publish ./ --publisher blueprints $*",
-    "blueprint:release": "yarn build:lib && yarn bump && yarn blueprint:synth --cache --clean-up false && yarn package && blueprint publish ./ --publisher blueprints $*",
+    "blueprint:preview": "yarn bump:preview && yarn blueprint:package && blueprint publish ./ --publisher private-blueprints-demo $*",
+    "blueprint:release": "yarn build:lib && yarn bump && yarn blueprint:synth --cache --clean-up false && yarn package && blueprint publish ./ --publisher private-blueprints-demo $*",
```
Comment (aggagen) [inline packages/blueprints/blueprint-builder/package.json:28]: Similar comment here. We should leave this as `blueprints`

File: packages/blueprints/blueprint-builder/package.json [hunk 3]
```diff
@@ -70,10 +70,10 @@
   ],
   "main": "lib/index.js",
   "license": "Apache-2.0",
-  "homepage": "https://aws.amazon.com/",
```
Comment (aggagen) [inline packages/blueprints/blueprint-builder/package.json:73]: there's a typo in this URL and the other `homepage`s below (missing the `:`)

File: packages/utils/projen-blueprint/src/blueprint.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import * as fs from 'fs';
 import * as path from 'path';
 import { typescript, javascript } from 'projen';
 import { generateTestSnapshotInfraFiles } from './test-snapshot';
+// import { hostname } from 'os';
```
Comment (aggagen) [inline packages/utils/projen-blueprint/src/blueprint.ts:5]: looks like this was accidentally added?

File: packages/utils/projen-blueprint/src/blueprint.ts [hunk 5]
```diff
@@ -102,6 +103,10 @@ export class ProjenBlueprint extends typescript.TypeScriptProject {
     };
     super(finalOpts);
 
+    this.addFields({
+      homepage: 'https//aws.amazon.com',
```
Comment (aggagen) [inline packages/utils/projen-blueprint/src/blueprint.ts:107]: I think we should add `homepage` as an optional parameter to the `ProjenBlueprintOptions` interface.

If the value is provided, then we add the `homepage` field to the `package.json`. If not, we do nothing (the current behavior). Otherwise, we'll start setting the homepage of all custom blueprints to this URL until your next PR to set this to the project URL in `blueprint-builder`.

Also, looks like the typo in the URL originates here: `https//aws.amazon.com`

File: packages/blueprints/blueprint-builder/src/build-blueprint-package.ts [hunk 6]
```diff
@@ -34,6 +34,7 @@ export function buildBlueprintPackage(
   });
 
   const newBlueprintOptions: ProjenBlueprintOptions = {
+    homepage: options.bpOptions.homepage,
```
Comment (aggagen) [inline packages/blueprints/blueprint-builder/src/build-blueprint-package.ts:37]: Is this `homepage` value actually used by the projen-blueprint? I don't see `homepage` defined in `ProjenBlueprintOptions`.

File: packages/blueprints/sam-serverless-app/static-assets/gradle/.idea/externalDependencies.xml
Comment (avi-alpert) [inline packages/blueprints/sam-serverless-app/static-assets/gradle/.idea/externalDependencies.xml:1]: None of the SAM files should be deleted

File: packages/utils/projen-blueprint/package.json [hunk 7]
```diff
@@ -23,23 +23,23 @@
     "npm:publish": "npm publish dist/js/*.tgz"
   },
   "devDependencies": {
-    "@amazon-codecatalyst/blueprint-util.cli": "*",
```
Comment (avi-alpert) [inline packages/utils/projen-blueprint/package.json:26]: Could we undo these changes to package.jsons? they should remain '*'

File: yarn.lock [hunk 8]
```diff
@@ -252,50 +252,50 @@ __metadata:
   languageName: unknown
   linkType: soft
 
-"@amazon-codecatalyst/blueprint-util.projen-blueprint@*, @amazon-codecatalyst/blueprint-util.projen-blueprint@^0.3.82, @amazon-codecatalyst/blueprint-util.projen-blueprint@^0.3.88, @amazon-codecatalyst/blueprint-util.projen-blueprint@workspace:packages/utils/projen-blueprint":
+"@amazon-codecatalyst/blueprint-util.projen-blueprint@*, @amazon-codecatalyst/blueprint-util.projen-blueprint@^0.3.147, @amazon-codecatalyst/blueprint-util.projen-blueprint@^0.3.82, @amazon-codecatalyst/blueprint-util.projen-blueprint@^0.3.88, @amazon-codecatalyst/blueprint-util.projen-blueprint@workspace:packages/utils/projen-blueprint":
```
Comment (avi-alpert) [inline yarn.lock:255]: I dont think there should be any changes to yarn.lock in this PR

**Review by avi-alpert — CHANGES_REQUESTED** [review]

**Review by avi-alpert — APPROVED** [review]

File: packages/blueprints/blueprint-builder/src/build-blueprint-package.ts [hunk 9]
```diff
@@ -20,20 +20,33 @@ import { Options } from './blueprint';
  * @param options
  */
 export function buildBlueprintPackage(
+  this: any,
```
Comment (avi-alpert) [inline packages/blueprints/blueprint-builder/src/build-blueprint-package.ts:23]: what is this for

**Review by avi-alpert — CHANGES_REQUESTED** [review]

**Review by avi-alpert — APPROVED** [review]
